src/main/java/org/truffleruby/core/encoding/EncodingNodes.java

```diff
@@ -699,18 +699,34 @@ protected int getIndex(RubyEncoding encoding) {
     // MRI: rb_enc_check_str / rb_encoding_check
     public abstract static class CheckRopeEncodingNode extends RubyBaseNode {
 
-        @Child private NegotiateCompatibleRopeEncodingNode negotiateCompatibleEncodingNode = NegotiateCompatibleRopeEncodingNode
-                .create();
-
         public static CheckRopeEncodingNode create() {
             return CheckRopeEncodingNodeGen.create();
         }
 
         public abstract RubyEncoding executeCheckEncoding(RopeWithEncoding first, RopeWithEncoding second);
 
-        @Specialization
+        @Specialization(guards = {
+                "first.getEncoding() == second.getEncoding()",
+                "first.getEncoding() == cachedEncoding"
+        })
+        protected RubyEncoding checkEncodingSameEncodingCached(RopeWithEncoding first, RopeWithEncoding second,
+                @Cached("first.getEncoding()") RubyEncoding cachedEncoding) {
+            return cachedEncoding;
+        }
+
+        @Specialization(guards = {
+                "first.getEncoding() == second.getEncoding()"
+        }, replaces = "checkEncodingSameEncodingCached")
+        protected RubyEncoding checkEncodingSameEncodingUncached(RopeWithEncoding first, RopeWithEncoding second,
+                @Cached("first.getEncoding()") RubyEncoding cachedEncoding) {
```

Comment:
Unused

Reply:
This code ended up being replaced by another commit. Do you want me to go through the history and fix its first occurrence? It'll conflict with the subsequent change, but I can deal with that if you'd like.